blogs/allSettled.md

```diff
@@ -0,0 +1,47 @@
+<a href="https://github.com/kriskowal/q">Q</a> has <a href="https://github.com/kriskowal/q/wiki/API-Reference#promiseallsettled">Q.allSettled</a>.  <a href="https://docs.angularjs.org/api/ng/service/$q">$q</a> does not.  I've noticed a lot of AngularJS devs request $q.allSettled.  So here it is!  <a href="http://jsfiddle.net/0sr65m7t/">JS Fiddle</a>
+
+<!--more-->
+<pre>
+angular.module("qImproved", [])
+.config(function ($provide) {
+  $provide.decorator("$q", function ($delegate) {
+    function allSettled(promises) {
+      var deferred = $delegate.defer(),
+          counter = 0,
+          results = angular.isArray(promises) ? [] : {};
+
+      angular.forEach(promises, function(promise, key) {
+        counter++;
+        $delegate.when(promise).then(function(value) {
+          if (results.hasOwnProperty(key)) return;
+          results[key] = { status: "fulfilled", value: value };
+          if (!(--counter)) deferred.resolve(results);
+        }, function(reason) {
+          if (results.hasOwnProperty(key)) return;
+          results[key] = { status: "rejected", reason: reason };
+          if (!(--counter)) deferred.resolve(results);
+        });
+      });
+
+      if (counter === 0) {
+        deferred.resolve(results);
```

Comment:
One difference between this and C#'s Task.WhenAll() is that it looks like the promise returned by allSettled will be fulfilled regardless of whether the underlying promises are fulfilled or rejected. The .NET API marks the wrapper task as faulted if any of the underlying tasks are faulted.

I'm not saying you should change the implementatin, but it's probably worth calling this out and why you like the behavior as is.